server/komodo-rest/src/main/java/org/komodo/rest/service/KomodoTeiidService.java

```diff
@@ -871,10 +1241,14 @@ public Response removeDriver(final @Context HttpHeaders headers,
 
             String title = RelationalMessages.getString(RelationalMessages.Info.DRIVER_DEPLOYMENT_STATUS_TITLE);
             KomodoStatusObject status = new KomodoStatusObject(title);
-            if (! hasDataSourceDriver(driverName, teiidInstance))
+            if (! hasDataSourceDriver(driverName, teiidInstance)) {
+                // Remove the Driver from cachedTeiid
+                CachedTeiid cachedTeiid = teiidNode.importContent(uow);
+                cachedTeiid.removeDriver(uow, driverName);
+                
```

Comment:
This doesn't make sense. As commented earlier, the CachedTeiid is just a cache. Why are we refreshing it then remove something from it? By L1248, the cache no longer reflects the state of the real Teiid instance so can no longer be called a cache.